### DISCUSSION TITLE
CLI argument to start in transposed layout?

### DISCUSSION THREAD

**Discussion by hi2u** [saulpw/visidata#2493] [Q&A]
90% of the time I want to view my data using the "transposed" (rotated) layout, so I have to hit shift-T to switch into it every time I run `vd`

Is there a CLI argument I can add to the `vs` shell command that sets which layout I want to start in?

**Comment by midichef** [comment]
One way that works would be to pipe your commands into visidata, and read them via `vd -p -`. The commands should be in .vdj format which is a JSONL file. In your case it would look like this:
```
echo '{"sheet": "", "col": "", "row": "", "longname": "transpose", "input": "", "keystrokes": null, "comment": null}' | \
    vd -p - -b -i datafile.tsv
```

To generate such .vdj commands in the future, do your commands in visidata:  `vd datafile.tsv`, press`T`, then press `Ctrl+D` to save the command log as a .vdj file. Then edit the .vdj file to replace any occurrences of `"sheet":"datafile.tsv"` with `"sheet":""`.

Alternatively, save the edited `transpose` .vdj command as `transpose.vdj` and run `vd -p transpose.vdj -b -i datafile.tsv`.

**Comment by ajkerrigan** [comment]
If in the vast majority of cases you want to automatically transpose on open, another option would be queueing a `transpose` anytime you open a sheet from a path, with a snippet like this in `~/.visidatarc`:

```python
from visidata import Sheet, vd

@Sheet.after
def afterLoad(sheet):
    if isinstance(sheet.source, vd.Path):
        vd.queueCommand('transpose', sheet=sheet)
```

Since `transpose` opens a new sheet, in that 10% of cases where you _don't_ want the transposed data you can `q` out of it. And since this only automatically transposes data when opening path-based sources, it won't get in your way as you open child sheets. (That also avoids some infinite loop risk)

**Comment by midichef** [comment]
Another option is to create a custom filetype `transpose`, that will load data from the file and then run transpose on it.
That way you can trigger transposing with `-f transpose`:    `vd -f transpose datafile.tsv`

To create the filetype for `transpose`, you just need to make a function `open_transpose()` in your `.visidatarc`, and tell it what the actual sheet type is that the underlying file should have, e.g. `TsvSheet`:
```
@VisiData.api
def open_transpose(vd, p):
    #change TsvSheet to be the Visidata sheet class for the actual file you're loading, like CsvSheet, JsonSheet, etc
    sheet_class = TsvSheet
    sheet = sheet_class(p.base_stem, source=p)
    vd.queueCommand('transpose', sheet=sheet)
    return sheet
```

You can find the sheet class by opening your file and inspecting the sheet with `=sheet.__class__`.
